### ISSUE TITLE
Code cleanup

### ISSUE THREAD

**Issue by bajtos** [strongloop-community/loopback-sdk-android#56]
See https://github.com/loopj/android-async-http/issues/751

1) We now support HEAD requests, 
https://github.com/strongloop/loopback-sdk-android/blob/4587dd82946c6a6647b60bd89ca74ad4ca66a481/src/main/java/com/strongloop/android/remoting/adapters/RestAdapter.java#L303

2) Your BinaryHandler and CallbackHandler override specific methods, check if you're overriding the ones you want (there are few variants of every callback method, depending on what type of information you do want/need about the request/response processing.

3) You could probably want to change declarations to require ResponseHandlerInterface instead of AsyncHttpResponseHandler

**Comment by yoshuacas** [comment]
Hi bajtos,

I noticed that you found this issue last year and tried to follow the discussion you had with the Loopj team, but I don't find a way to go around this issue, is there a way for me to work around it? I noticed you have it as an enhacement when in fact its breaking the sdk ugly.

I'm finding this when I try to PUT on my customized User Model, I'm simply trying to update a user property. I don't find it when doing a creation.

Thanks for the advice,

DC
